framework/include/utils/Conversion.h

```diff
@@ -75,6 +75,21 @@ std::string stringify(const SolveType & t);
 template <>
 std::string stringify(const ExecFlagType & t);
 
+/// Add no-op stringify if the argument already is a string
+template <>
+std::string stringify(const std::string & t);
```

Comment:
We have an "infix iterator" header: https://github.com/idaholab/moose/blob/devel/framework/include/utils/InfixIterator.h that allows you to do this kind of thing in a very generic context. @aeslaughter was also trying to add a `join()` method to MooseUtils, but with the iterator and the STL you can insert commas between items in any context pretty easily. 

We certainly don't need two (or any 😉).
https://github.com/idaholab/moose/pull/9242/files#diff-b667c5b178211dc59f0967396aeaf061R399